test/src/wasmJsMain/kotlin/io/github/irgaly/test/platform/Files.kt

```diff
@@ -0,0 +1,164 @@
+}
+
+private external interface FsSync: JsAny {
+    fun mkdtempSync(prefix: String): String
+}
+
+private external interface FileHandle: JsAny {
+    fun write(buffer: String, position: Int): Promise<WriteResult>
+    fun truncate(len: Int): Promise<Nothing?>
+    fun close(): Promise<Nothing?>
+}
+
+private external interface WriteResult: JsAny {
+    val bytesWritten: Int
+}
+
+private external interface Path: JsAny {
+    fun join(path1: String, path2: String): String
+}
+
+private external interface Os: JsAny {
+    fun tmpdir(): String
+}
+
+private external interface Error: JsAny {
+    val message: JsString
+}
+
+private fun fsRmOptions(): JsAny = js("({recursive: true, force: true})")
+
+actual class Files {
+    actual companion object {
+        actual suspend fun createTemporaryDirectory(): String = suspendCoroutine { continuation ->
+            if (isBrowser()) {
+                continuation.resume("js-browser-dummy-temporary-directory")
+            } else {
+                val fs = fs()
+                val path = path()
+                val os = os()
+                fs.mkdtemp(path.join(os.tmpdir(), "temp_")).then { directoryPath ->
+                    continuation.resume(directoryPath.toString())
+                    null
+                }.catch { error ->
+                    @Suppress("UNCHECKED_CAST_TO_EXTERNAL_INTERFACE")
+                    continuation.resumeWithException(IllegalStateException((error as Error).message.toString()))
+                    null
+                }
+            }
+        }
+
+        actual fun createTemporaryDirectorySync(): String {
+            return if (isBrowser()) {
+                "js-browser-dummy-temporary-directory"
+            } else {
+                val fsSync = fsSync()
+                val path = path()
+                val os = os()
+                fsSync.mkdtempSync(path.join(os.tmpdir(), "temp_"))
+            }
+        }
+
+        actual suspend fun createDirectory(path: String): Boolean =
+            suspendCoroutine { continuation ->
+                if (isBrowser()) {
+                    continuation.resume(true)
+                } else {
+                    val fs = fs()
+                    fs.mkdir(path).then {
+                        continuation.resume(true)
+                        null
+                    }.catch {
+                        continuation.resume(false)
+                        null
+                    }
+                }
+            }
+
+        actual suspend fun writeFile(path: String, text: String): Boolean =
+            suspendCoroutine { continuation ->
+                if (isBrowser()) {
+                    continuation.resume(true)
+                } else {
+                    val fs = fs()
+                    // change イベントのためになるべく上書きで更新する
+                    // * macOS ではこの処理でも rename イベントになってしまう
+                    fs.open(path, "r+").then<JsAny?> { handle ->
```

Comment:
* Promise then https://youtrack.jetbrains.com/issue/KT-19672/Promise-class-is-incorrect-in-the-Kotlin-runtime-for-the-JavaScript-platform